erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -905,14 +904,35 @@ def get_items(self):
 
 	def set_scrap_items(self):
 		if self.purpose != "Send to Subcontractor" and self.purpose in ["Manufacture", "Repack"]:
-			scrap_item_dict = self.get_bom_scrap_material(self.fg_completed_qty)
+			scrap_item_dict = {}
+			if frappe.db.get_single_value("Manufacturing Settings",	"backflush_scrap_items_based_on") =="BOM":
+				scrap_item_dict = self.get_bom_scrap_material(self.fg_completed_qty)
+			else:
+				if flt(self.fg_completed_qty) != flt(frappe.db.get_value("Work Order", self.work_order, 'qty')):
+					frappe.throw(_("For Qty must be same as Work Order Qty if backflush scrap items is based on Job Card."))
+				else:
+					scrap_item_dict = self.get_scrap_material()
 			for item in itervalues(scrap_item_dict):
 				item.idx = ''
 				if self.pro_doc and self.pro_doc.scrap_warehouse:
 					item["to_warehouse"] = self.pro_doc.scrap_warehouse
 
 			self.add_to_stock_entry_detail(scrap_item_dict, bom_no=self.bom_no)
 
+	def get_scrap_material(self):
+		scrap_items = {}
+		job_cards = frappe.get_all("Job Card", filters={'docstatus':1, 'work_order': self.work_order})
```

Comment:
Should this also account for Draft job cards? Also, small whitespace change:

```suggestion
		job_cards = frappe.get_all("Job Card", filters={'docstatus': 1, 'work_order': self.work_order})
```

Reply:
Only submitted stuff should show up in stock entries.